elastic/kibana
# [Security Solution] [Elastic AI Assistant] Fixes `ES|QL` codeblocks from not being able to be sent to Timeline

## Summary

Fixes `ES|QL` codeblocks from not being able to be sent to Timeline.

<p align="center">
  <img width="500" src="https://github.com/elastic/kibana/assets/2946766/fdc3b2a0-5b4b-4584-b304-c4d24de1917c" />
</p> 

## Test instructions

Either request the assistant to generate an ESQL query or just paste this codeblock into the conversation to test the action directly from the user message. Be sure to declare the codeblock language as `esql` or include one of the [string match patterns](https://github.com/elastic/kibana/pull/169478/files#diff-f70f0b96568e024e53bfbb62adcca72051f0a2e824d4ab22664eed0e149be248R38) above the code block so the action can be recognized.


````
Below is an `Elasticsearch Query Language` query:

```esql
FROM logs-endpoint*
| WHERE event.category == \"process\"
| STATS proc_count = COUNT(process.name) BY host.name
| KEEP host.name, proc_count
```
````


Note: The `send to timeline` actions appear to only reliably show up when using the Assistant instance within Timeline. Now that we have a more reliable way of attaching actions via markdown plugins/parsers, we should refactor this code to use that method as opposed to the code block/dom inspection route that is used currently. In the meantime I will see if there is a low-impact fix that can be made here.
